src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -252,6 +252,11 @@ private async Task ReplyOnReAuthenticationAsync<TIOAdapter>(TIOAdapter adapter,
         private async Task RenegotiateAsync<TIOAdapter>(TIOAdapter adapter)
             where TIOAdapter : IReadWriteAdapter
         {
+            if (_decryptedBytesCount is not 0)
```

Comment:
Are we allowed to read _decryptedBytesCount when we don't hold the locks?

Rather than moving this earlier, should it instead be moved down into the try?

Reply:
Putting it down would be safer. We try to prevent people from using this concurrently with IO so it should not be valid case but since it is easy I think it is ok to do little bit more work e.g. do the increments/decrements.

Reply:
moved to the try block